### PR TITLE
Vanotify - callback updates

### PR DESCRIPTION
**Pull request by nathanbwright**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES
- *(Summarize the changes that have been made to the platform)*: additional updates to our notification callback logic (behind a feature flag).
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*: VA Notify - yes, my team owns this module
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- https://github.com/orgs/department-of-veterans-affairs/projects/1415/views/1?pane=issue&itemId=85247150&issue=department-of-veterans-affairs%7Cvanotify-team%7C1411

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/va_notify/app/services/va_notify/status_update.rb [hunk 1]
```diff
@@ -28,7 +30,15 @@ def delegate(notification_callback)
...
 
     def constantized_class(class_name)
+      # notification.callback is set by other teams, not user input
       class_name.constantize
     end
+
+    def no_callback_info?
+      if notification.callback.blank?
+        Rails.logger.info(message: "VANotify - no callback provided for notification: #{notification.id}")
+        true
```
Comment (ianperera) [inline modules/va_notify/app/services/va_notify/status_update.rb:40]: What if the condition is not met? maybe return false if not met?

File: modules/va_notify/app/services/va_notify/status_update.rb [hunk 2]
```diff
@@ -28,7 +30,15 @@ def delegate(notification_callback)
     private
 
     def constantized_class(class_name)
+      # notification.callback is set by other teams, not user input
       class_name.constantize
     end
+
+    def no_callback_info?
```
Comment (ianperera) [inline modules/va_notify/app/services/va_notify/status_update.rb:37]: What do you think about renaming to `callback_info_missing`?

**Review by MarchandMD — APPROVED** [review]
Approving on this assumption of understanding for `caller_locations`: 

caller_locations is an array of invocations, similar to a stack trace, and the explicit `return` will be from the location whence called from, outside of the 3 filepaths listed

**Review by ianperera — APPROVED** [review]
LGTM

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**